packages/eslint-plugin/src/rules/require-await.ts

```diff
@@ -75,13 +84,108 @@ export default createRule({
         !isEmptyFunction(node) &&
         !(scopeInfo.isGen && scopeInfo.isAsyncYield)
       ) {
+        let fix: ReportFixFunction | undefined;
+
+        // If the function belongs to a method definition or
+        // property, then the function's range may not include the
+        // `async` keyword and we should look at the parent instead.
+        const nodeWithAsyncKeyword =
+          (node.parent.type === AST_NODE_TYPES.MethodDefinition &&
+            node.parent.value === node) ||
+          (node.parent.type === AST_NODE_TYPES.Property &&
+            node.parent.method &&
+            node.parent.value === node)
+            ? node.parent
+            : node;
+
+        const asyncToken = context.sourceCode.getFirstToken(
+          nodeWithAsyncKeyword,
+          token => token.value === 'async',
+        );
+
+        if (asyncToken) {
+          const nextTokenWithComments = context.sourceCode.getTokenAfter(
+            asyncToken,
+            { includeComments: true },
+          );
+
+          if (nextTokenWithComments) {
+            const asyncRange: Readonly<AST.Range> = [
+              asyncToken.range[0],
+              nextTokenWithComments.range[0],
+            ] as const;
+
+            // Removing the `async` keyword can cause parsing errors if the
+            // current statement is relying on automatic semicolon insertion.
+            // If ASI is currently being used, then we should replace the
+            // `async` keyword with a semicolon.
+            const nextToken = context.sourceCode.getTokenAfter(asyncToken);
+            const addSemiColon =
+              nextToken?.type === AST_TOKEN_TYPES.Punctuator &&
+              (nextToken.value === '[' || nextToken.value === '(') &&
+              (nodeWithAsyncKeyword.type === AST_NODE_TYPES.MethodDefinition ||
+                isStartOfExpressionStatement(nodeWithAsyncKeyword)) &&
+              needsPrecedingSemicolon(context.sourceCode, nodeWithAsyncKeyword);
+
+            const changes = [
+              { range: asyncRange, replacement: addSemiColon ? ';' : '' },
+            ];
+
+            // If there's a return type annotation and it's a
```

Comment:
Let's _not_ do this if the function is a generator function (see also comment on test file), since it doesn't make semantic sense. 

We _could_ consider doing `AsyncGenerator` -> `Generator` if provided... I leave it to you as an option whether to a) do nothing with the return type if specified on a generator, or b) change the type if and only if it's an `AsyncGenerator`.  Please add a test case for whichever route you go 🙂. 

PS - today I learned that this rule inspects generator functions 🙃.

Reply:
I choose to go with replacing `AsyncGenerator` with `Generator` since it was easy enough to do.